src/Microsoft.AspNetCore.Mvc.Core/Internal/MiddlewarePipelineResourceFilter.cs

```diff
@@ -0,0 +1,64 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Filters;
+
+namespace Microsoft.AspNetCore.Mvc.Internal
+{
+    public class MiddlewarePipelineResourceFilter : IAsyncResourceFilter, IOrderedFilter
```

Comment:
This isn't really a problem. The same limitations apply today using middleware in general.